pinot-common/src/main/java/org/apache/pinot/common/utils/RegexpPatternConverterUtils.java

```diff
@@ -18,16 +18,23 @@
  */
 package org.apache.pinot.common.utils;
 
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+
 /**
  * Utility for converting regex patterns.
  */
 public class RegexpPatternConverterUtils {
   private RegexpPatternConverterUtils() {
   }
 
-  /* Represents all metacharacters to be processed */
-  public static final String[] REGEXP_METACHARACTERS =
-      {"\\", "^", "$", ".", "{", "}", "[", "]", "(", ")", "*", "+", "?", "|", "<", ">", "-", "&", "/"};
+  /*
+   * Represents all metacharacters to be processed.
+   * This excludes the \ (back slash) character as that doubles up as an escape character as well.
+   * So it is handled separately in the conversion logic.
+   */
+  public static final Set<String> REGEXP_METACHARACTERS = ImmutableSet.of(
```

Comment:
Why the move to a set? Since the number of characters is small, isn't an array more space efficient?

Reply:
Ack, will fix this.